crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -45,7 +51,76 @@ impl_downcast!(Stage);
 ///
 /// The checker may report a system more times than the amount of constraints it would actually need
 /// to have unambiguous order with regards to a group of already-constrained systems.
-pub struct ReportExecutionOrderAmbiguities;
+///
+/// By default only a warning with the number of unresolved ambiguities detected will be reported per [`SystemStage`].
+/// This behavior can be changed by explicitly adding this resource using the following constructors:
+/// * [ReportExecutionOrderAmbiguities::off()] - Disables all messages reported by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::minimal()] - Displays only the number of unresolved ambiguities detected by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::verbose()] - Displays a full report of ambiguities detected by the ambiguity checker.
+///
+/// The ambiguity checker will ignore all crates starting with "bevy_". To ignore a custom crate, just call [`ReportExecutionOrderAmbiguities::ignore`] function
+/// with an list of crates names as an argument.
+///
+/// ## Example
+/// ```ignore
+/// # use bevy_app::App;
+/// # use bevy_ecs::schedule::ReportExecutionOrderAmbiguities;
+/// App::new()
+///    .insert_resource(ReportExecutionOrderAmbiguities::verbose().ignore(&["my_external_crate"]));
+/// ```
+pub struct ReportExecutionOrderAmbiguities {
+    pub level: AmbiguityReportLevel,
+    pub ignore_crates: Vec<String>,
+}
+
+impl ReportExecutionOrderAmbiguities {
+    /// Disables all messages reported by the ambiguity checker.
+    pub fn off() -> Self {
+        Self {
+            level: AmbiguityReportLevel::Off,
+            ..Default::default()
+        }
+    }
+
+    /// Displays only the number of unresolved ambiguities detected by the ambiguity checker. This is the default behavior.
+    pub fn minimal() -> Self {
+        Self {
+            level: AmbiguityReportLevel::Minimal,
+            ..Default::default()
+        }
+    }
+
+    /// Displays a full report of ambiguities detected by the ambiguity checker.
+    pub fn verbose() -> Self {
+        Self {
+            level: AmbiguityReportLevel::Verbose,
+            ..Default::default()
+        }
+    }
+
+    /// Adds the given crates to be ignored by ambiguity checker. Check [`ReportExecutionOrderAmbiguities`] for more details.
+    pub fn ignore(mut self, create_prefix: &[&str]) -> Self {
+        for s in create_prefix {
+            self.ignore_crates.push(s.to_string());
+        }
+        self
+    }
+
+    #[doc(hidden)]
+    pub fn include_bevy(mut self) -> Self {
+        self.ignore_crates.retain(|s| !s.starts_with("bevy_"));
+        self
+    }
+}
+
+impl Default for ReportExecutionOrderAmbiguities {
+    fn default() -> Self {
+        Self {
+            level: AmbiguityReportLevel::Minimal,
+            ignore_crates: vec!["bevy_".into()],
```

Comment:
I dont like the `bevy_` Filtering here.
Looking at the [Bevy Assets](https://bevyengine.org/assets/) the vast majority of the Plugins start with `bevy_` or `bevy-` (which inside the sourcecode is stored as `bevy_`).
This means that Plugin Authors could potentialy miss ambiguities because they missed this Default.

Yes there is the `include_bevy()` Function that removes the Filter, but it is hiddden in the docs and a User might think it only targets offical bevy crates.

I would suggest that in addition of this Blacklist, `ReportExecutionOrderAmbiguities` should also have a Whitelist that takes priority.

Reply:
Instead of increasing the complexity with blocklist and allowlist, I think it's better to hard code all bevy internal crates. I agree that this may confuse users, since the main goal is to ignore bevy internal crates, not third party with the similar names.

The downside of hard coding bevy internal crates is that it'll require additional maintenance when new crates are added. We can list also bevy reserved names crates, to reduce the maintenance, but it'll be require in the future.

Let's see if someone else have others toughts.

Reply:
I would argue that plugin authors confident enough to create something within `bevy_` "namespace" are likely proficient enough to know about this, but I realise that this is wishful thinking.

A while ago I suggested an alternative solution: shove all internal systems into a common ambiguity set. This will solve the problem, but re-enabling ambiguity detection for those systems for Bevy development will be harder.

A better solution is to go over everything and diligently assign ambiguity sets (and system-pair ambiguities) where it makes sense. Cart all but vetoed this, because that would be noisy. I agree, but it is the best we can do if we only do what is strictly correct.

Yet another option: remove the fuzzy `bevy_` filter, add a number of strict filters, one per official crate, like `bevy_app`, `bevy_render`, etc.  This leaves unofficial plugins alone, at which point they can resolve ambiguities in whichever way they want, including adding their own filter during plugin initialization. In fact, official plugins could add their filters this way, too.

Reply:
> Yet another option: remove the fuzzy bevy_ filter, add a number of strict filters, one per official crate, like bevy_app, bevy_render, etc.

I think this is the way to go, as much as I would like to fully resolve all of the ambiguities in the engine today.

Reply:
So I'll add by default, a list with all oficial and reserved bevy crates by default, that way, third party crates won't be hidden by default.

Reply:
Addressed this on latest commit